spec/controllers/idv/confirmations_controller_spec.rb

```diff
@@ -0,0 +1,80 @@
+require 'rails_helper'
+
+describe Idv::ConfirmationsController do
+  render_views
+
+  let(:user) { create(:user, :signed_up, email: 'old_email@example.com') }
+  let(:applicant) { Proofer::Applicant.new first_name: 'Some', last_name: 'One' }
+  let(:agent) { Proofer::Agent.new vendor: :mock }
+  let(:resolution) { agent.start applicant }
+
+  describe 'before_actions' do
+    it 'includes before_actions from AccountStateChecker' do
+      expect(subject).to have_actions(
+        :before,
+        :confirm_two_factor_authenticated
+      )
+    end
+  end
+
+  describe 'all questions answered correctly' do
+    it 'shows success' do
+      init_idv_session
+      complete_idv_session(true)
+
+      get :index
+
+      expect(response.status).to eq 200
+      expect(response.body).to include(t('idv.titles.complete'))
+    end
+  end
+
+  describe 'some answers incorrect' do
+    it 'shows error' do
+      init_idv_session
+      complete_idv_session(false)
+
+      get :index
+
+      expect(response.status).to eq 200
+      expect(response.body).to include(t('idv.titles.hardfail'))
+    end
+  end
+
+  describe 'questions incomplete' do
+    it 'redirects to /idv/questions' do
+      init_idv_session
+
+      get :index
+
+      expect(response).to redirect_to(idv_questions_path)
+    end
+  end
```

Comment:
Could all 4 of the above specs be wrapped in a `context 'session already started'`.. with `before { init_idv_session }`?